### ISSUE TITLE
Interpolation is too lazy

### ISSUE THREAD

**Issue by kaan-atakan** [vinissimus/next-translate#781]
Here is a simple example:

```js
t(
  "key",
  { something: "something", somethingElse: "something else" },
  { default: "{{somethingElse}}" }
)
```

The result of the function call above is `something` when it should be `something else`. The correct value is returned when the data payload does not contain a shorter match:

```js
t(
  "key",
  { aThing: "a thing", somethingElse: "something else" },
  { default: "{{somethingElse}}" }
)
```

This is almost certainly an issue with one of the regexes in interpolate@transCore, my attempts at fixing it myself have failed.

**Comment by kaan-atakan** [comment]
I was actually able to fix it, but I'm not sure how to make a PR. Here is my version of interpolate:

```tsx
function interpolation({
  text,
  query,
  config,
  lang,
}: {
  text?: string
  query?: TranslationQuery | null
  config: I18nConfig
  lang?: string | undefined
}): string {
  if (!text || !query) return text || ''

  const escapeRegex = (str: string) =>
    str.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&')
  const {
    format = null,
    prefix = '{{',
    suffix = '}}',
  } = config.interpolation || {}

  const regexEnd =
    suffix === '' ? '' : `([\\s,]+([\\w-]*))?\\s*${escapeRegex(suffix)}`
  return Object.keys(query).reduce((all, varKey) => {
    const regex = new RegExp(
      `${escapeRegex(prefix)}\\s*${varKey}${regexEnd}`,
      'gm'
    )
    // $1 is the first match group
    return all.replace(regex, (_match, $1, $2) => {
      // $1 undefined can mean either no formatting requested: "{{name}}"
      // or no format name given: "{{name, }}" -> ignore
      return $2 && format
        ? (format(query[varKey], $2, lang) as string)
        : (query[varKey] as string)
    })  
  }, text)
}
```

All the tests seem to run fine.

**Comment by kaan-atakan** [comment]
The gist of the changes are the ending regex is changed from:

```regex
\\s*,?\\s*([\\w-]+)?\\s*
```
to 

```regex
([\\s,]+([\\w-]*))?\\s*
```

And replace gets another parameter $2, which is used in lieu of $1

**Comment by aralroca** [comment]
Thanks for your PR. Released on [1.3.4](https://github.com/vinissimus/next-translate/releases/tag/1.3.4)
